drivers: flash: spi_nor: fix driver initialization error

xinyangt9 opened the pull request:
Fix the issue that the initialization will fail when both CONFIG_SPI_NOR_SFDP_RUNTIME and CONFIG_SPI_NOR_IDLE_IN_DPD are enabled. The cause of this problem is simply calling the wrong function.

Fixes #33015

Signed-off-by: Xinyang Tan <xinyang.tan@delve.com>

jasonxytan commented:
@nvlsianpu Hi. Would you mind approving running the remaining 5 workflows? Thank you!

Commit:
drivers: flash: spi_nor: fix driver initialization error

Fix the issue that the initialization will fail when both
CONFIG_SPI_NOR_SFDP_RUNTIME and CONFIG_SPI_NOR_IDLE_IN_DPD are enabled.
The cause of this problem is simply calling the wrong function.

Fixes #33015

Signed-off-by: Xinyang Tan <xinyang.tan@delve.com>

nvlsianpu commented:
Patch is appropriate.

I think initialization might be reworked so (especially by reworking release_device() a little) eneter_dpd might be called only once, at the end.
